### PR TITLE
DNM: Fix up rpm builds; silence some verbosity

### PR DESCRIPTION
**Pull request by dmick**

### CODE REVIEW

File: build-ceph-rpm.sh [hunk 1]
```diff
@@ -13,8 +13,9 @@ VER="$(git describe)"
 
 # Reformat version if needed to match RPM version and release
 if expr index $(git describe --always) '-' > /dev/null ; then
-    RPM_VER=$(git describe --always | cut -d'-' -f1)
-    RPM_REL=$(git describe --always | cut -d- -f2- | tr '-' '.')
+    desc = $(git describe --always | sed 's/^v//'
```
Comment (tchaikov) [inline build-ceph-rpm.sh]: missing a `)`

File: build-ceph-rpm.sh [hunk 2]
```diff
@@ -54,7 +55,7 @@ echo --START-IGNORE-WARNINGS
 # we only need to use autogen here if we need a dist tarball
 if [ -x make-dist ]; then
     echo --STOP-IGNORE-WARNINGS
-    ./make-dist
+    ./make-dist $RPM_VER
```
Comment (tchaikov) [inline build-ceph-rpm.sh]: might want to pass `$VER` to `make-dist` instead?

Comment (liewegas) [inline build-ceph-rpm.sh]: it needs to be the short version to make rpmbuild happy.

**Comment by dmick** [comment]
Good catch @tchaikov, I was rushing to reproduce what I'd already tested on the gb.  diffed this time to be sure.

File: build-ceph-rpm.sh [hunk 3]
```diff
@@ -13,8 +13,9 @@ VER="$(git describe)"
 
 # Reformat version if needed to match RPM version and release
 if expr index $(git describe --always) '-' > /dev/null ; then
-    RPM_VER=$(git describe --always | cut -d'-' -f1)
-    RPM_REL=$(git describe --always | cut -d- -f2- | tr '-' '.')
+    desc = $(git describe --always | sed 's/^v//')
```
Comment (tchaikov) [inline build-ceph-rpm.sh]: might want to remove the spaces around `=`.

File: build-ceph-rpm.sh (hunk 2 above)
Comment (tchaikov) [inline build-ceph-rpm.sh]: got it. but we will have the identical VERSION and RPM_RELEASE in the `ceph.spec`. currently they will both be "11.0.0". is this expected?

Comment (tchaikov) [inline build-ceph-rpm.sh]: maybe we need to update "make-dist" to address this?

File: build-ceph-rpm.sh (hunk 3 above)
Comment (dmick) [inline build-ceph-rpm.sh]: OY!  Yes, you're right

File: build-ceph-rpm.sh (hunk 2 above)
Comment (dmick) [inline build-ceph-rpm.sh]: shouldn't be; RPM_VER and RPM_REL are different

**Comment by dmick** [comment]
Added another commit because the last two new sed's were not having any effect.  I'll squash before merge if this makes sense.

File: build-ceph-rpm.sh [hunk 4]
```diff
@@ -101,6 +102,10 @@ if [[ "$rpm_version" =~  .*-rc[1-9]$ ]] ; then
     fi
     sed -i "s/^Source0:.*/Source0:        http:\/\/ceph.com\/download\/%{name}-%{version}-$rpm_rc.tar.bz2/" ceph.spec
     sed -i "s/^%setup.*/%setup -q -n %{name}-%{version}-$rpm_rc/" ceph.spec
+else
+    # fix up spec file for non-rc builds too
+    sed -i "s/^Version:.*/Version:        $RPM_VER/" ceph.spec
+    sed -i "s/^Release:.*/Release:        $RPM_REL/" ceph.spec
```
Comment (tchaikov) [inline build-ceph-rpm.sh]: cool. this fixes the "Release"!

**Comment by tchaikov** [comment]
lgtm.

**Comment by tchaikov** [comment]
@dmick http://tracker.ceph.com/issues/16501 is the tracker ticket related to your fix.

**Pull request merged**